### ISSUE TITLE
Throws an errror “Cannot read property 'file' of undefined”

### ISSUE THREAD

**Issue by shinnn** [gucong3000/postcss-jsx#26]
```js
'use strict';
const postcssJsx = require('postcss-jsx'); // v0.32.0

postcssJsx.parse("let a;let a;", {
  from: 'source.ts'
});
```

```console
$ node run.js
TypeError: Cannot read property 'file' of undefined
    at Scope.checkBlockScopedCollisions (/Users/shinnn/example/node_modules/@babel/traverse/lib/scope/index.js:347:22)
    at Scope.registerBinding (/Users/shinnn/example/node_modules/@babel/traverse/lib/scope/index.js:506:16)
    at Scope.registerDeclaration (/Users/shinnn/example/node_modules/@babel/traverse/lib/scope/index.js:454:14)
    at Object.Declaration (/Users/shinnn/example/node_modules/@babel/traverse/lib/scope/index.js:125:12)
    at NodePath._call (/Users/shinnn/example/node_modules/@babel/traverse/lib/path/context.js:53:20)
    at NodePath.call (/Users/shinnn/example/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/shinnn/example/node_modules/@babel/traverse/lib/path/context.js:88:12)
    at TraversalContext.visitQueue (/Users/shinnn/example/node_modules/@babel/traverse/lib/context.js:118:16)
    at TraversalContext.visitMultiple (/Users/shinnn/example/node_modules/@babel/traverse/lib/context.js:85:17)
    at TraversalContext.visit (/Users/shinnn/example/node_modules/@babel/traverse/lib/context.js:144:19)
```

**Comment by shinnn** [comment]
> the underlying issue upstream has been resolved and released I believe.

@ntwb As described above, no. I'm surprised that one of the stylelint maintainer joined the FUD on Twitter. Eh?

**Comment by ntwb** [comment]
@shinnn Honestly I have no idea if that was FUD or not, the closet reference I was able to find in your repo that this was an upstream issue referenced #8, that was fixed ~3 months ago.

I'm unable to determine myself what the current issue is, is it this new issue that you have just created?

----

As an aside and sorry for being off topic here @gucong3000 as all issues are locked on @shinnn's repo I'm making this comment here.

Instead of closing all those tickets @shinnn as a duplicate of the exact issue number of the issue it is a duplicate of, the issue is closed against an issue that itself has already been closed, it's very confusing, for example:

• 231 & 234 are closed as duplicates of 222
• 222 is closed as a duplicate of 173
• 173 is closed as a duplicate of 148
• 148 is closed linking to itself in 148

None of the above makes easily determining what the issue is, let alone what the duplciate is, I honestly have no idea and cannot determine it myself.

Each of the above issues should have been closed and marked as duplcates of 148, not each of those tickets, there are lots more examples of similar issues.

@shinnn you do not have to allow comments on your repo, that is your personal choice but if I may make one suggestion, if you are getting multiple issues that are duplcates of existing issues for whatever reason users are not able to determine that, this could possibly be handled by adding some more docs or an FAQ to help explain these common issues users are having with the extension.

p.s Thanks for the extension, I use it all the time and love it ❤️
